iree/modules/check/test/BUILD

```diff
@@ -21,6 +21,13 @@ package(
     licenses = ["notice"],  # Apache 2.0
 )
 
+iree_cmake_extra_content(
+    content = """
+if(${IREE_BUILD_COMPILER} AND ${IREE_BUILD_TESTS})
```

Comment:
Hmmm I'd rather put this in the rules themselves so we don't have to have this custom stuff all over. iree_lit_test should already noop if `IREE_BUILD_TESTS` is false. Technically we could use lit tests for something that doesn't require the compiler, but I don't think we ever will, so I'd be fine just calling this a compiler tool as well and making it require `IREE_BUILD_COMPILER`

Reply:
+1, add `IREE_BUILD_COMPILER` to these checks instead of adding this logic to a few places:

https://github.com/google/iree/blob/f9c0862cbefce22d47c974ddb690d59240495aaa/build_tools/cmake/iree_lit_test.cmake#L33-L36

https://github.com/google/iree/blob/f9c0862cbefce22d47c974ddb690d59240495aaa/build_tools/cmake/iree_lit_test.cmake#L110-L113

https://github.com/google/iree/blob/main/build_tools/cmake/iree_check_test.cmake (as needed)

The `iree/modules/tensorlist/`, `iree/modules/strings`, and `iree/vm` changes will stay, since those use `cc_test`s depending on `iree_bytecode_module` output. Any places that just use `iree_lit_test`/`iree_check_test` can do the filtering implicitly through the rules.

Reply:
Done